src/lib/operators/join_hash.cpp

```diff
@@ -46,11 +46,13 @@ std::shared_ptr<const Table> JoinHash::_on_execute() {
   /*
   This is the expected implementation for swapping tables:
   (1) if left or right outer join, outer relation becomes probe relation (we have to swap only for left outer)
+  (2) for an anti join the inputs are always swapped
   (2) else the smaller relation will become build relation, the larger probe relation
   (3) for full outer joins we currently don't have an implementation.
   */
-  if (_mode == JoinMode::Left || (_mode != JoinMode::Right &&
-                                  (_input_left->get_output()->row_count() > _input_right->get_output()->row_count()))) {
+  if (_mode == JoinMode::Left || _mode == JoinMode::Anti ||
+      (_mode != JoinMode::Right &&
+       (_input_left->get_output()->row_count() > _input_right->get_output()->row_count()))) {
```

Comment:
This conditional is getting too complex for me. I think it would be a good idea to mix it with the comment

```
bool some_flag = false;

//   (1) if left or right outer join, outer relation becomes probe relation (we have to swap only for left outer)
if (...) some_flag = true;

//   (2) for an anti join the inputs are always swapped
if (_mode == JoinMode::Anti) some_flag = true

if (some_flag) {...}
```